packages/lit-html/src/directives/render-light.ts

```diff
@@ -0,0 +1,106 @@
+import {Directive, DirectiveResult, directive, NodePart} from '../lit-html.js';
+
+export interface RenderLightHost extends HTMLElement {
+  renderLight(): unknown;
+}
+
+class RenderLight extends Directive {
+  render() {
+    /* SSR handled specially in render-lit-html */
+  }
+  update(part: NodePart) {
+    const instance = part.parentNode as RenderLightHost;
+    if (typeof instance.renderLight === 'function') {
+      return instance.renderLight();
```

Comment:
Hmm, but then calling that callback would need to be part of `LitElement`'s lifecycle right? Right now they're totally decoupled, which seems good.

I think you can argue `renderLight` is a replacement for the container scope actually rendering DOM into the children of a CE; updates to CE child DOM does normally go before the async update of the CE and you could probably argue that's a _good_ thing, given that children are sort of one type of input to a CE.

Reply:
I added an issue capturing the outcome of our discussion on this: https://github.com/Polymer/lit-html/issues/1387. Since it's slightly meaty, we'll do it separately.